ra2ce/graph/origins_destinations.py

```diff
@@ -455,93 +545,92 @@ def get_node_id_from_position(
+            # and it goes to the except statement.
+            try:
+                closest_u_v_k = inverse_vertices_dict[
+                    (closest_node_on_road[0], closest_node_on_road[1])
+                ]
+                match_edge = graph.edges[closest_u_v_k]
+
+                # Add the node to the graph edge
+                match_geom = match_edge["geometry"]
+
+                new_lines = split_line_with_points(match_geom, [match_od_point])
+                if len(new_lines) == 1:
+                    inverse_nodes_dict = find_closest_node(
+                        closest_node_on_road,
+                        inverse_vertices_dict,
+                        inverse_nodes_dict,
+                        graph,
+                    )
+                    assert len(new_lines) == 2
+                    assert len([match_od_point]) == 1
+
+                    line1, line2 = new_lines
+
+                    new_node_id = max_node_id + 1
+                    max_node_id = new_node_id
+
+                    graph, updated_new_node_ids = update_or_add_node(
+                        graph, new_node_id, match_od_point, match_name, category, od_data, od, i
+                    )
+
+                    for updated_new_node_id in updated_new_node_ids:
+                        if updated_new_node_id != closest_u_v_k[0] and updated_new_node_id != closest_u_v_k[1]:
+                            # Update the inverse_nodes_dict with the new node
+                            if match_od_point.coords[0] in inverse_nodes_dict:
+                                if not updated_new_node_id in inverse_nodes_dict[match_od_point.coords[0]]:
+                                    inverse_nodes_dict[match_od_point.coords[0]].append(updated_new_node_id)
+                            else:
+                                # If the key doesn't exist, create a new list with updated_new_node_id
+                                inverse_nodes_dict[match_od_point.coords[0]] = [updated_new_node_id]
+
+                            # Delete the new node from the inverse_vertices_dict as no end-points are included here
+                            del inverse_vertices_dict[match_od_point.coords[0]]
+
+                            # Update the graph edges and the inverse_vertices_dict
+                            graph, inverse_vertices_dict = update_edges_with_new_node(
+                                graph,
+                                match_edge,
+                                closest_u_v_k[0],
+                                closest_u_v_k[1],
+                                closest_u_v_k[2],
+                                line1,
+                                line2,
+                                updated_new_node_id,
+                                crs,
+                                inverse_vertices_dict,
+                            )
+
+            except (KeyError, AssertionError):
```

Comment:
I'm not a fan of having `try-except`, the second one (`AssertionError`) can be avoided with relative small code modifications, but how is the first one ( `KeyError`) triggered?